system/jlib/jflz.cpp

```diff
@@ -637,11 +640,13 @@ class jlib_decl CFastLZCompressor : public CInterface, public ICompressor
         if (trailing)
             return;
         size32_t toflush = (inlenblk==COMMITTED)?inlen:inlenblk;
+        if (toflush == 0)
+            return;
         assertex(outlen+sizeof(size32_t)*2+toflush+fastlzSlack(toflush)<=blksz);
         size32_t *cmpsize = (size32_t *)(outbuf+outlen);
         byte *out = (byte *)(cmpsize+1);
         *cmpsize = (size32_t)fastlz_compress(inbuf, (int)toflush, out, ht);
-        if (*cmpsize<=toflush) {
+        if (*cmpsize<toflush) {
```

Comment:
>2). If compressed output is same size as input then do not save compressed data.

Is this related to here?
What were the consequences of doing so, inefficiency only?
Or was it ultimately the cause of the 'CMemoryBufferSerialStream::get read past end of stream error' errors ?